nwbinspector/register_checks.py

```diff
@@ -0,0 +1,139 @@
+        `from nwbinspector.register_checks import Severity`. A good example is comparing if h5py.Dataset compression
+        has been enabled on smaller vs. larger objects (see nwbinspect/checks/nwb_containers.py for details).
+
+        The user will never directly see this severity, but it will prioritize the order in which check results are
+        presented by the NWBInspector.
+    """
+
+    message: str
+    severity: Severity = None
+    importance: Importance = Importance.BEST_PRACTICE_SUGGESTION
+    check_function_name: str = ""
+    object_type: str = ""
+    object_name: str = ""
+    location: str = ""
+
+
+def register_check(importance, neurodata_type):
+    """Wrap a check function to add it to the list of default checks for that severity and neurodata type."""
+
+    def register_check_and_auto_parse(check_function):
+        if importance not in Importance:
+            raise ValueError(
+                f"Indicated importance ({importance}) of custom check ({check_function.__name__}) is not a valid "
+                "importance level! Please choose one of Importance.CRITICAL, Importance.BEST_PRACTICE_VIOLATION, "
+                "or Importance.BEST_PRACTICE_SUGGESTION."
+            )
+        check_function.importance = importance
+        check_function.neurodata_type = neurodata_type
+
+        @wraps(check_function)
+        def auto_parse_some_output(*args, **kwargs):
+            if args:
+                obj = args[0]
+            else:
+                obj = kwargs[list(kwargs)[0]]
+
+            auto_parsed_result = check_function(*args, **kwargs)
+            if auto_parsed_result is not None:
+                if auto_parsed_result.severity is None:  # For perfect consistency with not specifying
+                    auto_parsed_result.severity = Severity.NO_SEVERITY
+                if auto_parsed_result.severity not in Severity:
+                    raise ValueError(
+                        f"Indicated severity ({auto_parsed_result.severity}) of custom check "
+                        f"({check_function.__name__}) is not a valid severity level! Please choose one of "
+                        "Severity.HIGH, Severity.LOW, or do not specify any severity."
+                    )
+
+                auto_parsed_result.importance = check_function.importance
+                auto_parsed_result.check_function_name = check_function.__name__
+                auto_parsed_result.object_type = type(obj).__name__
+                auto_parsed_result.object_name = obj.name
+                auto_parsed_result.location = parse_location(nwbfile_object=obj)
+            return auto_parsed_result
+
+        available_checks[check_function.importance][check_function.neurodata_type].append(auto_parse_some_output)
+
+        return auto_parse_some_output
+
+    return register_check_and_auto_parse
+
+
+def parse_location(nwbfile_object):
```

Comment:
```suggestion
def parse_location(nwbfile: : NWBFile):
```

Just for consistency, I would suggest renaming the parameter to ``nwbfile`` and adding the type hint.